src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -84,6 +113,20 @@ trait EntityComponent {
       }
     }
 
+    def unmarshalEntities(entityAttributeAction: ReadAction[Seq[EntityListResult]]): ReadAction[Iterable[Entity]] = {
+      entityAttributeAction.map { entityAttributeRecords =>
+        val entityRecords = entityAttributeRecords.map(r => EntityRecord(r.entityId, r.entityName, r.entityType, r.workspaceId)).toSet
+        val attributesByEntityId = attributeQuery.unmarshalAttributes[Long](entityAttributeRecords.collect {
```

Comment:
in general, would love to see more comments. If you recall, I asked about why you're using `collect` here (and now I understand) ... always more comments!